test/e2e/l7networkpolicy_test.go

```diff
@@ -434,12 +441,35 @@ type L7LogEntry struct {
 	DestPort    int32            `json:"dest_port"`
 	Protocol    string           `json:"proto"`
 	AppProtocol string           `json:"app_proto,omitempty"`
+	Packet      string           `json:"packet"`
 	Http        *L7LogHttpEntry  `json:"http,omitempty"`
 	Alert       *L7LogAlertEntry `json:"alert,omitempty"`
 }
 
+// Compares if two L7LogEntry are equal. They must be exactly equal if Packet is
+// in the same base. Otherwise, if either of the Packet is in base64, they equal
+// if Packet in base64 contains essential message in other Packet.
 func (e *L7LogEntry) Equal(x *L7LogEntry) bool {
-	return reflect.DeepEqual(e, x)
+	packetLooseEqual := conversion.EqualitiesOrDie(
+		func(a, b L7LogEntry) bool {
+			if a != b {
```

Comment:
that doesn't really seem useful here, as it can never be true?

Reply:
Actually it is needed, for the second case when both are bytes.
```
// ignore unexpected log entries and duplicates
if slices.ContainsFunc(expected, log.Matches) && !slices.ContainsFunc(actual, log.Matches) {
```